pkg/types/chain_reader.go

```diff
@@ -0,0 +1,56 @@
+package types
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc/status"
+)
+
+type errChainReader string
```

Comment:
The google style guide recomends an `Error` suffix for types:
```suggestion
type chainReaderError string
```